src/merkle/store/mod.rs

```diff
@@ -219,6 +232,55 @@ impl MerkleStore {
         })
     }
 
+    // LEAF ACCESSORS
+    // --------------------------------------------------------------------------------------------
+
+    /// Returns the leaf at `index` rooted on the tree `root`.
+    ///
+    /// # Errors
+    ///
+    /// This method can return the following errors:
+    /// - `RootNotInStore` if the `root` is not present in the store.
+    /// - `NodeNotInStore` if a node needed to traverse from `root` to `index` is not present in the store.
+    /// - `DepthTooBig` if `leaf_index.depth` is too big.
+    /// - `DepthTooSmall` if the final node is not a leaf.
+    pub fn get_leaf(&self, root: Word, leaf_index: NodeIndex) -> Result<Word, MerkleError> {
```

Comment:
Didn't we deprecate the concept of leaf? Is it expected to be a pre-image of the node value?

Reply:
I'm not sure, @bobbinth do you have comments?